### PR TITLE
Feature - Include additional linters for golangci

### PR DESCRIPTION
**Pull request by naveensrinivasan**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes/features)
N/A

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Linting #173 

* **What is the current behavior?** (You can also link to an open issue here)
It does not have additional lighters enabled.

* **What is the new behavior (if this is a feature change)?**
Feature - Include additional linters for golangci. This is settings is based on https://github.com/kubernetes-sigs/security-profiles-operator/blob/master/.golangci.yml

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
None

* **Other information**:

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,13 +15,14 @@ tidy:
 
 GOLANGCI_LINT = $(shell pwd)/bin/golangci-lint
 golangci-lint:
+	rm -rf bin || :
 	@[ -f $(GOLANGCI_LINT) ] || { \
```
Comment (inferno-chromium) [inline Makefile:19]: This file exists check can be removed then if we are removing the dir/file anyway at start.

File: Makefile [hunk 2]
```diff
@@ -15,13 +15,14 @@ tidy:
 
 GOLANGCI_LINT = $(shell pwd)/bin/golangci-lint
 golangci-lint:
+	rm -rf bin || :
```
Comment (inferno-chromium) [inline Makefile:18]: Why remove entire dir, why not just remove the file e.g. rm -f $(GOLANGCI_LINT) ?

**Review by inferno-chromium — APPROVED** [review]
LGTM with a change needed.

File: Makefile (hunk 2 above)
Comment (naveensrinivasan) [reply Makefile:18]: I agree we can just remove the `rm -f $(GOLANGCI_LINT)`

Comment (naveensrinivasan) [reply Makefile]: Took care of it!

File: Makefile (hunk 1 above)
Comment (naveensrinivasan) [reply Makefile]: Took care of it! Thanks

**Pull request merged**